scripts/setup_dev_env.sh

```diff
@@ -0,0 +1,200 @@
+HAAUKINS_EXERCISES_REPO="git@github.com:aau-network-security/haaukins-exercises.git"
+VIRTUAL_DEV_ENV="git@github.com:aau-network-security/sec0x.git"
+
+PROGRAMS=(go vagrant packer)
+REPOS=($HAAUKINS_REPO $HAAUKINS_STORE_REPO $HAAUKINS_EXERCISES_REPO $HAAUKINS_WEBCLIENT_REPO)
+
+PROJECT_DIR=$HOME/Documents/project
+
+mkdir -p $PROJECT_DIR
+
+cd $PROJECT_DIR
+
+ # clone repositories
+ for i in ${REPOS[@]}; do
+     echo "Cloning repository $i"
+     git clone $i
+ done
+
+
+
+# install golang 
+INSTALL_GOLANG_MACOS() {
+    which -s go
+    if [[ $? != 0 ]] ; then
+       # Install golang
+        echo "Installing GoLang..."
+        wget https://go.dev/dl/go1.18.1.darwin-amd64.pkg
+        sudo installer -pkg go1.18.1.darwin-amd64.pkg -target ~/Applications/
+
+    else
+      echo "Go is already installed !!"
+      exit 1 
+    fi
+}
+
+
+# install homebrew
+INSTALL_HOMEBREW() {
+
+    which -s brew
+    if [[ $? != 0 ]] ; then
+       # Install Homebrew
+        echo "Installing homebrew..."
+        "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+    else
+        echo "Homebrew is already installed, updating brew."
+        brew update
+    fi
+}
+
+# install vagrant
+INSTALL_VAGRANT_MACOS () {
+    which -s vagrant
+    if [[ $? != 0 ]] ; then
+       # Install Vagrant
+       echo "Installing vagrant through brew"
+       brew install vagrant
+    else
+        echo "vagrant is already installed"
+    fi
+
+}
+
+
+# install packer
+INSTALL_PACKER_MACOS() {
+    which -s packer
+    if [[ $? != 0 ]] ; then
+       # Install Packer
+        echo "Installing packer through brew"
+        brew tap hashicorp/tap
+        brew install hashicorp/tap/packer
+    else
+        echo "Packer is already installed"
+      #  brew upgrade hashicorp/tap/packer
+    fi
+}
+
+BUILD_VM_ENV() {
+    echo "Building VM environment"
+    cd $PROJECT_DIR/sec0x/hkn-base
+    ./build.sh
+    cd $PROJECT_DIR/sec0x
+    # make sure plugins are installed
+    echo "Installing plugin: vagrant-disksize"
+    vagrant plugin install vagrant-disksize
+    echo "Installing plugin: vagrant-vbguest"
+    vagrant plugin install vagrant-env
+    echo "Starting machine with vagrant up"
+    vagrant up
+}
+
+INSTALL_GAWK () {
+  which -s gawk
+    if [[ $? != 0 ]] ; then
+       echo "gawk is already installed"
+    else
+        # Install Gawk
+        echo "Installing gawk through brew"
+        brew install gawk
+    fi
+}
+
+
+INSTALL_VIRTUALBOX_DEBIAN() {
+
+    if which vboxmanage >/dev/null; then
+        echo "vbox is already installed"
+    else
+        echo "Installing vboxmanage ... "
+        sudo apt update 
+        curl https://download.virtualbox.org/virtualbox/6.1.8/Oracle_VM_VirtualBox_Extension_Pack-6.1.8.vbox-extpack --output Oracle_VM_VirtualBox_Extension_Pack-6.1.8.vbox-extpack 
+        sudo VBoxManage extpack install Oracle_VM_VirtualBox_Extension_Pack-6.1.8.vbox-extpack  
+        sudo apt install virtualbox -y
```

Comment:
Switch around? does it work when vbox isn't installed?

Reply:
Also why 6.1.8?

Reply:
There is no specific reason of having version 6.1.8, would be nice to have latest stable version, which seems 6.1.34 we have on official virtualbox page. Updated it now.

> Also why 6.1.8?

Reply:
> Switch around? does it work when vbox isn't installed?

Could you expand what you meant @Lanestolen, I could not understand the part "Switch around? " .

Reply:
So vbox is installed before the extension package,
Also it seems like there is issues installing vbox 6.1.34 on Ubuntu 20.04

Reply:
Yeah vbox needs to be installed before extension package. 
Hmm, I did not notice the issues, I will check it through

Reply:
Oh yeah, my bad. Thanks a lot for reviewing it.
I will switch the order, Virtualbox needs to be installed before extension package, right. 
Then check the issues regarding to Installing vbox 6.1.34.

I am thinking to create a ticket which checks functions inside this script on CI pipeline. 
What do you think ?  Basically, it will call functions on CI pipeline and check whether installation of any package creates an issue or not. 
Cloning sec0x and making it up might not be feasible on CI but for others it seems feasible to do.

Reply:
Sound cool : )